src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -84,5 +84,82 @@ public void Dictionary_Generic_RemoveKey_DefaultKeyContainedInDictionary(int cou
         }
 
         #endregion
+
+        #region EnsureCapacity
+        
+        [Fact]
+        public void EnsureCapacity_NegativeCapacityRequested_Throws()
+        {
+            var dictionary = new Dictionary<string, string>();
+            Assert.Throws<ArgumentOutOfRangeException>(() => dictionary.EnsureCapacity(-1));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(3)]
+        public void EnsureCapacity_DefaultCapacityOnEmptyDictionary_ReturnsCapacityRequestedLargerOrEqual(int requestedCapacity)
+        {
+            var dictionary = new Dictionary<string, string>();
+            Assert.True(requestedCapacity <= dictionary.EnsureCapacity(requestedCapacity));
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(3)]
+        public void EnsureCapacity_CapacityRequestedSmallerThanCurrentCapacity_CapacityUnchanged(int requestedCapacity)
+        {
+            var dictionary = new Dictionary<string, string>();
+            var capacity = dictionary.EnsureCapacity(requestedCapacity);
+            Assert.Equal(capacity, dictionary.EnsureCapacity(requestedCapacity - 1));
+        }
+        
+        [Theory]
+        [InlineData(1)]
+        [InlineData(5)]
+        public void EnsureCapacity_CapacityRequestedSmallerThanCount_SetsCapacityToLargerOrEqualToExistingCount(int count)
+        {
+            var dictionary = new Dictionary<int, int>();
+            for (int i = 0; i < count; i++)
+            {
+                dictionary.Add(i, i);
+            }
+            Assert.True(dictionary.Count <= dictionary.EnsureCapacity(count - 1));
```

Comment:
Ditto; such comparisons are better with InRange because it provides more actionable output of the assert fails.

Reply:
Thanks. I changed the assertion to use InRange instead. 
Since I'm only looking to check the lower bound, I set the upper upper bound in Assert.InRange to int.MaxValue.

Reply:
@ViktorHofer recommends I add GreaterThan and LessThan helper methods to AssertExtensions. 

I'll do that now and refresh this PR afterwards.

Reply:
Correction:

As @ViktorHofer suggested I also checked for Assert.GreaterThan and Assert.LessThan.

Since there is no such apis, then as @stephentoub also suggested I am using the universal Assert.InRange(a,b,c) the "GreaterThan" check